soccer/planning/Path.hpp

```diff
@@ -18,6 +25,9 @@ class Path {
     Path() : _startTime(0) {}
     virtual ~Path() {}
 
+    /// Returns true if the path is non-empty
+    virtual bool valid() const { return true; }
```

Comment:
Right now it's possible to construct a CompositePath or an InterpolatedPath that's empty.  The `valid()` check just ensures that the path is an actual path with content.